packages/did-provider-ethr/src/ethr-did-provider.ts

```diff
@@ -25,43 +116,93 @@ export function toEthereumAddress(hexPublicKey: string): string {
+    ttl?: number
+    /**
+     * @deprecated Please use the `networks` parameter instead.
+     */
+    network?: string | number
+    /**
+     * @deprecated Please use the `networks` parameter instead.
+     */
+    name?: string
+    /**
+     * @deprecated Please use the `networks` parameter instead.
+     */
     rpcUrl?: string
+    /**
+     * @deprecated Please use the `networks` parameter instead.
+     */
     web3Provider?: Provider
-    ttl?: number
-    gas?: number
+    /**
+     * @deprecated Please use the `networks` parameter instead.
+     */
     registry?: string
+    /**
+     * @deprecated If tweaking is necessary, please specify the maximum `gasLimit` as an `option` to every DID update
+     */
+    gas?: number
   }) {
     super()
     this.defaultKms = options.defaultKms
-    this.network = options.network
-    this.rpcUrl = options.rpcUrl
-    this.web3Provider = <Provider>options.web3Provider
-    if (typeof this.web3Provider === 'undefined') {
-      this.web3Provider = new JsonRpcProvider(this.rpcUrl, this.network)
+    if (options.networks) {
+      this.networks = options.networks
+    } else {
+      const singleNetwork: EthrNetworkConfiguration = {
+        provider: options.web3Provider,
+        rpcUrl: options.rpcUrl,
+        registry: options.registry,
+      }
+      if (typeof singleNetwork.provider === 'undefined') {
+        singleNetwork.provider = new JsonRpcProvider(singleNetwork.rpcUrl, singleNetwork.network)
+      }
+      if (typeof options.network === 'string') {
+        if (options.network.startsWith('0x')) {
+          singleNetwork.chainId = parseInt(options.network.substring(2), 16)
+        } else {
+          singleNetwork.name = options.network
+        }
+      } else if (typeof options.network === 'number') {
+        singleNetwork.chainId = options.network
+        singleNetwork.name = options.name
+      }
+      this.networks = [singleNetwork]
     }
     this.ttl = options.ttl
     this.gas = options.gas
-    this.registry = options.registry
   }
 
   async createIdentifier(
-    { kms, options }: { kms?: string; options?: any },
+    { kms, options }: { kms?: string; options?: CreateDidEthrOptions },
     context: IRequiredContext,
   ): Promise<Omit<IIdentifier, 'provider'>> {
     const key = await context.agent.keyManagerCreate({ kms: kms || this.defaultKms, type: 'Secp256k1' })
     const compressedPublicKey = computePublicKey(`0x${key.publicKeyHex}`, true)
+    let networkSpecifier =
+      options?.network ||
+      (options?.providerName?.match(/^did:ethr:.+$/) ? options?.providerName?.substring(9) : undefined)
+    const network = this.getNetworkFor(networkSpecifier)
+    if (!network) {
+      throw new Error(
+        `invalid_config: Cannot create did:ethr. There is no known configuration for network=${networkSpecifier}'`,
+      )
+    }
+    if (typeof networkSpecifier === 'number') {
+      console.log('opting for new network')
```

Comment:
@mirceanis you probably forgot to remove this debug `console.log`

Reply:
great catch!